src/Repository/ConnectedGithubClient.php

```diff
@@ -0,0 +1,195 @@
+<?php
+
+namespace OAT\DependencyResolver\Repository;
+
+use Github\Exception\ErrorException;
+use Github\Exception\RuntimeException;
+use OAT\DependencyResolver\Repository\Entity\Repository;
+use OAT\DependencyResolver\Repository\Exception\BranchNotFoundException;
+use OAT\DependencyResolver\Repository\Exception\EmptyRepositoryException;
+use OAT\DependencyResolver\Repository\Exception\FileNotFoundException;
+
+class ConnectedGithubClient
+{
+    /** @var GithubClientProxy */
+    protected $client;
+
+    /** @var string */
+    protected $organization = '';
+
+    /** @var string */
+    protected $token = '';
+
+    /**
+     * Is the client authenticated?
+     *
+     * @var bool
+     */
+    protected $authenticated = false;
+
+    /**
+     * Stores the numbers of public and private repositories.
+     *
+     * @var array
+     */
+    protected $organisationProperties = [];
+
+    public function __construct(GithubClientProxy $client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * Stores token for github client.
+     */
+    public function setToken(string $token)
+    {
+        $this->token = $token;
+    }
+
+    public function getOrganizationProperties(string $owner): array
+    {
+        $this->authenticateAndCheck($owner);
+
+        return $this->organisationProperties;
+    }
+
+    /**
+     * Retrieves organization's repositories.
+     *
+     * @param string $owner
+     * @param int    $perPage Number of repositories to load per client call.
+     *
+     * @return array
+     */
+    public function getRepositoryList(string $owner, int $perPage = 100): array
+    {
+        $this->authenticateAndCheck($owner);
+
+        $repositories = [];
+        $page = 0;
+        do {
+            $page++;
+            $foundRepositories = $this->client->getRepositoryList($owner, $page, $perPage);
+            foreach ($foundRepositories as $repository) {
+                $repositoryName = $repository['name'];
+                $repositories[$owner . '/' . $repositoryName] = new Repository(
+                    false, // repository not yet analyzed.
```

Comment:
Flag parameters are code smells. What does exactly mean `analyzed`? We should not even able to instantiate invalid entities, right? It should always be in a valid state.

Reply:
This flag just says: I found a repository in the list and I didn't run through it to find all its properties (manifest and composer presence, etc). It is then used when exploring the repositories one by one to avoid running this again and again on the same repositories. The entity is then valid and persisted to the global map, but not yet filled up with all the things we'll later find for each repository.